src/components/LMap.vue

```diff
@@ -5,18 +5,22 @@
 </template>
 
 <script setup lang="ts">
-import { onBeforeUnmount, ref, watch } from 'vue';
+import { onBeforeUnmount, shallowRef, watch } from 'vue';
 import 'leaflet/dist/leaflet.css';
-import { map, icon, marker, tileLayer} from 'leaflet';
+import { map, icon, marker, tileLayer, type Marker} from 'leaflet';
 import { useI18n } from 'vue-i18n';
+import { usePermission } from '@vueuse/core';
 import Azul from '@/assets/pin/Pin_Azul.png';
 import Verde from '@/assets/pin/Pin_Verde.png';
 import Rojo from '@/assets/pin/Pin_Rojo.png';
 import Dot from '@/assets/pin/dot.png';
 import { useToast } from '@/composables/use-toast';
-import { isNumber } from '@/utils/validation';
+import { isNil, isNumber } from '@/utils/validation';
 
 const props = defineProps<{ markers: MapMarker[] }>();
+
+const locationAccess = usePermission('microphone');
```

Comment:
```suggestion
const locationAccess = usePermission('geolocation');
```

No? 😆

Si te da error de tipos, prueba de todas maneras a ver si en tiempo de ejecución detecta bien el watcher los cambios que das en la autorización y ya intento yo hacer un PR a la librería para arreglar los tipos.